dir-index.html

```diff
@@ -39,36 +40,60 @@
     </div>
   </div>
   <div id="content">
-      <div id="content-header">
-        <strong>Index of {{ .Path }}</strong>
+    <div id="content-header" class="d-flex flex-wrap">
+      <div>
+        <strong>
+          Index of
+          {{ range .Breadcrumbs -}}
+          /{{ if .Path }}<a href="{{ $root.GatewayURL }}{{ .Path | urlEscape }}">{{ .Name }}</a>{{ else }}{{ .Name }}{{ end }}
+          {{- else }}
+          {{ .Path }}
+          {{ end }}
+        </strong>
         {{ if .Hash }}
         <div class="ipfs-hash">
           {{ .Hash }}
         </div>
         {{ end }}
       </div>
-      <table>
-        <tr>
-          <td class="type-icon">
-            <div class="ipfs-_blank">&nbsp;</div>
-          </td>
-          <td>
-            <a href="{{.BackLink | urlEscape}}">..</a>
-          </td>
-          <td></td>
-        </tr>
-        {{ range .Listing }}
-        <tr>
-          <td class="type-icon">
-            <div class="{{iconFromExt .Name}}">&nbsp;</div>
-          </td>
-          <td>
-            <a href="{{ .Path | urlEscape }}">{{ .Name }}</a>
-          </td>
-          <td class="no-linebreak">{{ .Size }}</td>
-        </tr>
+      {{ if .Size }}
+      <div class="no-linebreak flex-shrink-1 ml-auto">
+        <strong>&nbsp;{{ .Size }}</strong>
+      </div>
       {{ end }}
-      </table>
+    </div>
+    <div class="table-responsive">
+    <table>
+      <tr>
+        <td class="type-icon">
+          <div class="ipfs-_blank">&nbsp;</div>
+        </td>
+        <td>
+          <a href="{{.BackLink | urlEscape}}">..</a>
+        </td>
+        <td></td>
+        <td></td>
+      </tr>
+      {{ range .Listing }}
+      <tr>
+        <td class="type-icon">
+          <div class="{{iconFromExt .Name}}">&nbsp;</div>
+        </td>
+        <td>
+          <a href="{{ .Path | urlEscape }}">{{ .Name }}</a>
+        </td>
+        <td class="no-linebreak">
+          {{ if .Hash }}
+          <a href="{{ $root.GatewayURL }}/ipfs/{{ .Hash | urlEscape }}">
+            {{ .ShortHash }}
+          </a>
+          {{ end }}
+        </td>
```

Comment:
@jessicaschilling thoughts on making CIDs monospace and use the same color as CID in the header to not distract from actual listing?

I think we could extend & reuse `ipfs-hash` class here – made a quick mockup:

![2020-07-22--01-33-06](https://user-images.githubusercontent.com/157609/88117451-788c3980-cbbb-11ea-8600-21d026431c0d.png)


(AFAIK this is not a part of the Bounty, so could be a separate PR)

Reply:
I'll do this in a separate PR - need to port changes over to dir-index-uncat anyway, so can do these in tandem.